lib/synapse/haproxy.rb

```diff
@@ -1030,19 +1030,33 @@ def generate_backend_stanza(watcher, config)
       ]
     end
 
+    def haproxy_exec(command)
+      info = nil
+      s = UNIXSocket.new(@opts['socket_file_path'])
+      s.write(command)
+      info = s.read
+    rescue StandardError => e
```

Comment:
Ambivalent about it. I would lean towards having a helper method that does not trigger the restart in that case to keep it grouped where possible, but I'll pull up + reduplicate the rescue code for now.